### PR TITLE
Creation of a spinner component (#3163)

### PR DESCRIPTION
**Pull request by olivierPigeon-RTE**
Signed-off-by: olivierPigeon-RTE <olivier.pigeon@rte-france.com>

- In release note :
  -  In chapter :  Tasks
  -  Text : #3163 : Creation of a spinner component and spinners added in real time users screen and in activity area screen

### CODE REVIEW

File: ui/main/src/app/modules/share/spinner/spinner.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+<!-- file, You can obtain one at http://mozilla.org/MPL/2.0/.              -->
+<!-- SPDX-License-Identifier: MPL-2.0                                      -->
+<!-- This file is part of the OperatorFabric project.                      -->
+
+<div *ngIf="mustDisplaySpinner" class="opfab-loading">
+    <div id="opfab-loading-spinner" class="opfab-loading-spinner">
+        <em class="fas fa-spinner fa-spin opfab-slow-spinner"></em>
+    </div>
+    <div style="text-align: center;font:16px;font-weight: bold;margin-top:10px">
+       {{ getTranslation(loadingText) }}
```
[the remark this answers is not in the document]
Comment (olivierPigeon-RTE) [reply ui/main/src/app/modules/share/spinner/spinner.component.html:14]: it is actually needed, {{loadingText}} doesn't display the text properly

File: ui/main/src/app/modules/archives/archives.component.html [hunk 2]
```diff
@@ -112,18 +112,7 @@
...
-  <em  class="fas fa-spinner fa-spin opfab-slow-spinner" ></em>
-  </div>
-  <div style="text-align: center;font-size:16px;font-weight: bold;margin-top:10px" translate>
-      archive.loadingInProgress
-  </div>
-
-  <br/>
-  <br/>
-</div>
+<of-spinner *ngIf="loadingInProgress" loadingText="archive.loadingInProgress"></of-spinner>
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/archives/archives.component.html:115]: use the generic message here and remove translation key archive.loadingInProgress

File: ui/main/src/app/modules/share/spinner/spinner.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }, this.timeBeforeDisplayingSpinner);
+    }
+
+    ngOnDestroy() {
+        clearInterval(this.interval);
+    }
+
+    getTranslation(stringToTranslate: string): string {
+        return this.translate.instant(stringToTranslate);
+    }
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/share/spinner/spinner.component.ts:39]: not needed anymore

File: ui/main/src/app/modules/share/spinner/spinner.module.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+import {NgModule} from '@angular/core';
+import {CommonModule} from '@angular/common';
+import {TranslateModule} from '@ngx-translate/core';
+import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
+import { SpinnerComponent } from './spinner.component';
+
+@NgModule({
+    imports: [CommonModule, TranslateModule, NgbModule],
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/share/spinner/spinner.module.ts:17]: Translate module shall not be needed anymore due to previous review comment

File: ui/main/src/app/modules/realtimeusers/realtimeusers.component.ts [hunk 5]
```diff
@@ -27,12 +28,15 @@ export class RealtimeusersComponent implements OnInit, OnDestroy {
...
 
     realTimeScreens: Array<RealTimeScreen>;
-    realTimeScreensLoaded = false;
+    isRealTimeScreensLoaded = false;
     realTimeScreenIndexToDisplay: number;
     connectedUsersPerEntityAndGroup: Map<string, Array<string>> = new Map<string, Array<string>>();
     realTimeScreensOptions = [];
     columnsNumberPerScreenAndScreenColumn: Map<string, number> = new Map<string, number>();
 
+    @ViewChild('spinner') customSpinner: SpinnerComponent;
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/realtimeusers/realtimeusers.component.ts:37]: I do not see any use of it ?

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Comment by freddidierRTE** [comment]
rebase is needed to solve conflict

File: src/test/cypress/cypress/integration/1-RealTimeUsers.spec.js [hunk 6]
```diff
@@ -11,6 +11,14 @@
...
 
+    function checkLoadingSpinnerIsDisplayed() {
+        cy.get('#opfab-loading-spinner').should('exist');
+    }
+
+    function checkLoadingSpinnerIsNotDisplayed() {
+        cy.get('#opfab-loading-spinner').should('not.exist');
+    }
+
     before('Set up configuration', function () {
```
Comment (freddidierRTE) [inline src/test/cypress/cypress/integration/1-RealTimeUsers.spec.js:22]: Add this two method in cypress command.js to avoid duplication

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/share/spinner/spinner.module.ts (hunk 4 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/share/spinner/spinner.module.ts:17]: yes you are right

File: src/test/cypress/cypress/integration/1-RealTimeUsers.spec.js [hunk 7]
```diff
@@ -215,4 +215,18 @@ describe ('RealTimeUsersPage',()=>{
...
         cy.get('table').first().find('tr').eq(4).find('td').eq(0).should('contain.text', '(operator4_fr)');
     })
+
+
+    it ('Check spinner is displayed when request is delayed and that spinner disappears once the request arrived ', function () {
+        cy.delayRequestResponse('/businessconfig/realtimescreens');
+
+        cy.loginOpFab('operator1_fr','test');
+
+        // We move to logging screen
```
Comment (freddidierRTE) [inline src/test/cypress/cypress/integration/1-RealTimeUsers.spec.js:225]: remove invalid comment

File: src/test/cypress/cypress/integration/ActivityArea.spec.js [hunk 8]
```diff
@@ -235,4 +235,19 @@ describe ('ActivityAreaPage',()=>{
 
         cy.setPropertyInConf('selectActivityAreaOnLogin ','web-ui',false);
     })
+
+    it ('Check spinner is displayed when request is delayed and that spinner disappears once the request arrived ', function () {
+        cy.delayRequestResponse('/users/users/*');
+
+        cy.loginOpFab('operator1_fr','test');
+
+        // We move to logging screen
```
Comment (freddidierRTE) [inline src/test/cypress/cypress/integration/ActivityArea.spec.js:244]: remove invalid comment

File: ui/main/src/app/modules/share/spinner/spinner.component.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    styleUrls: ['./spinner.component.scss']
+})
+export class SpinnerComponent implements OnInit, OnDestroy {
+    @Input() loadingText = "shared.loadingInProgress";
+    @Input() timeBeforeDisplayingSpinner = 500;
+
+    mustDisplaySpinner = false;
+    private interval;
+
+    constructor(protected translate: TranslateService) {}
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/share/spinner/spinner.component.ts:25]: translate service not needed here

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Pull request merged**